h2o-genmodel/src/main/java/hex/genmodel/tools/PrintMojo.java

```diff
@@ -13,34 +13,53 @@
 import java.io.IOException;
 import java.io.PrintStream;
 import java.util.*;
+import java.util.List;
+
+import static water.util.JavaVersionUtils.JAVA_VERSION;
 
 /**
  * Print dot (graphviz) representation of one or more trees in a DRF or GBM model.
  */
-public class PrintMojo {
-  
+public class PrintMojo implements MojoPrinter {
   enum Format {
-    dot, json, raw
+    dot, json, raw, png
   }
   
-  private MojoModel genModel;
-  private Format format = Format.dot;
-  private int treeToPrint = -1;
-  private int maxLevelsToPrintPerEdge = 10;
-  private boolean detail = false;
-  private String outputFileName = null;
-  private String optionalTitle = null;
-  private PrintTreeOptions pTreeOptions;
-  private boolean internal;
+  protected MojoModel genModel;
+  protected Format format = Format.dot;
+  protected int treeToPrint = -1;
+  protected int maxLevelsToPrintPerEdge = 10;
+  protected boolean detail = false;
+  protected String outputFileName = null;
+  protected String optionalTitle = null;
+  protected PrintTreeOptions pTreeOptions;
+  protected boolean internal;
+  protected final String tmpOutputFileName = "tmpOutputFileName.gv";
 
   public static void main(String[] args) {
+    MojoPrinter mojoPrinter = null;
+    
+    if ("png".equals(getFormat(args))) {
+      if (JAVA_VERSION.isKnown() && JAVA_VERSION.getMajor() > 7) {
+        ServiceLoader<MojoPrinter> mojoPrinters = ServiceLoader.load(MojoPrinter.class);
+        for (MojoPrinter printer : mojoPrinters) {
+          if ("hex.genmodel.tools.JgraphtPrintMojo".equals(printer.getClass().getName())) {
```

Comment:
this can break in case of refactoring, what about a supportsFormat(Format) method instead?
woudl also get rid of the if below
basically I would refactor this to a findPrinterForFormat() method that would fail on JDK7 for PNG this way you can get rid of the check in validateArgs

Reply:
fixed as discussed